Core/OxidEEEvents.php

```diff
@@ -407,26 +395,37 @@ public static function onActivate()
     public static function onDeactivate()
     {
         self::$oDb = DatabaseProvider::getDb();
-        // read the OXID_ENVIRONMENT variable from the .env and docker-compose files
-        $environmentVar = getenv('OXID_ENVIRONMENT');
-
-        // if development, delete the database tables on module de-activation
-        if ($environmentVar === 'development') {
-            self::_deleteOrderTransactionTable();
-        }
 
-        self::_disablePaymentTypes();
+        self::_disablePaymentMethods();
     }
 
     /**
      * Deactivate wirecard payment methods
      *
      * @since 1.0.0
      */
-    private static function _disablePaymentTypes()
+    private static function _disablePaymentMethods()
     {
         $sQuery = "UPDATE oxpayments SET `OXACTIVE` = 0 WHERE `OXID` LIKE 'wd%'";
 
         self::$oDb->execute($sQuery);
     }
+
+    /**
+     * Migration steps from version 1.0.0 to 1.0.1
+     *
+     * @since 1.0.1
+     */
+    private static function _migrateFrom100To101()
```

Comment:
We might want to create a migration class to handle all migration logic and initialize it on startup. Otherwise we'd potentially end up with lots of migration functions like this scattered across all files.

Not sure if this should be done in this PR though.

Reply:
there will not be that many releases that have a database migration. Let's refactor this once it's starting to get a problem. 🤔

Reply:
it is fine for now
some migration steps might have to be called in different places and time of execution